packages/amplify-ui-components/src/components/amplify-require-new-password/amplify-require-new-password.tsx

```diff
@@ -0,0 +1,109 @@
+import { Component, Prop, State, h } from '@stencil/core';
+import { FormFieldTypes } from '../amplify-auth-fields/amplify-auth-fields-interface';
+import { AuthState, ChallengeName, CognitoUserInterface } from '../../common/types/auth-types';
+import { CHANGE_PASSWORD, CHANGE, BACK_TO_SIGN_IN, NO_AUTH_MODULE_FOUND } from '../../common/constants';
+// import { CodeDeliveryType } from './amplify-forgot-password-interface';
+
+import { Auth } from '@aws-amplify/auth';
+import { Logger, isEmpty } from '@aws-amplify/core';
+
+const logger = new Logger('amplify-require-new-password');
+
+@Component({
+  tag: 'amplify-require-new-password',
+  shadow: false,
+})
+export class AmplifyRequireNewPassword {
+  /** The header text of the forgot password section */
+  @Prop() headerText: string = CHANGE_PASSWORD;
+  /** The text displayed inside of the submit button for the form */
+  @Prop() submitButtonText: string = CHANGE;
+  /** The form fields displayed inside of the forgot password form */
+  @Prop() formFields: FormFieldTypes;
+  /** (Optional) Overrides default styling */
+  @Prop() overrideStyle: boolean = false;
+  /** The function called when submitting a new password */
+  @Prop() handleSubmit: (event: Event) => void = event => this.change(event);
+  /** Passed from the Authenticatior component in order to change Authentication state */
+  @Prop() handleAuthStateChange: (nextAuthState: AuthState, data?: object) => void;
+
+  /** Used for the username to be passed to resend code */
+  @Prop() user: CognitoUserInterface;
+
+  @State() username: string;
+  @State() password: string;
+  @State() code: string;
+
+  componentWillLoad() {
+    this.formFields = [
+      {
+        type: 'password',
+        required: true,
+        handleInputChange: event => this.handlePasswordChange(event),
```

Comment:
I finally understand why there are so many anonymous, scoped closures in these components:


```suggestion
        handleInputChange: this.handlePasswordChange,
```

The handler functions themselves aren't scoped:

```js
handlePasswordChange(event) {
  this.password = event.target.value;
}

// vs.
handlePasswordChange = (event) => {
  this.password = event.target.value;
}
```

If they were, then almost all extraneous closures could be removed, except for situtations where the event & the handler don't share 1-to-1 positional parameters:

```js
handleInputChange: event => this.updatePassword(event.target.value)
```

Reply:
This is a common scenario for JSX/React users, and even has a mention regarding preferences & performance here:
> https://reactjs.org/docs/handling-events.html

Nothing specific to change: this is primarily subjective unless performance data shows otherwise.  Personally, I'm disappointed that `class` methods aren't automatically scoped to `this`, which is a major stumbling point for newcomers to JavaScript.

Reply:
agreed. I will add a note to investigate the performance difference here. Do you have a strong concern against the current use of anonymous, scoped closures?

Reply:
The anon function usage we have in our code is based off of this guide:
https://stenciljs.com/docs/forms#advanced-forms

Reply:
No concerns without explicit data to justify the change. 

Just pointing out TIL moments for me :)